bee-storage/bee-storage-rocksdb/src/storage.rs

```diff
@@ -0,0 +1,225 @@
+        opts.set_env(&env);
+
+        let db = DB::open_cf_descriptors(
+            &opts,
+            config.path,
+            vec![
+                cf_system,
+                cf_message_id_to_message,
+                cf_message_id_to_metadata,
+                cf_message_id_to_message_id,
+                cf_index_to_message_id,
+                cf_output_id_to_created_output,
+                cf_output_id_to_consumed_output,
+                cf_output_id_unspent,
+                cf_ed25519_address_to_output_id,
+                cf_ledger_index,
+                cf_milestone_index_to_milestone,
+                cf_snapshot_info,
+                cf_solid_entry_point_to_milestone_index,
+                cf_milestone_index_to_output_diff,
+                cf_address_to_balance,
+                cf_milestone_index_to_unconfirmed_message,
+                cf_milestone_index_to_receipt,
+                cf_spent_to_treasury,
+            ],
+        )?;
+
+        let mut flushopts = FlushOptions::new();
+        flushopts.set_wait(true);
+        db.flush_opt(&flushopts)?;
+        db.flush_cf_opt(db.cf_handle(CF_SYSTEM).unwrap(), &flushopts)?;
+
+        Ok(db)
+    }
+
+    pub(crate) fn cf_handle(&self, cf_str: &'static str) -> Result<&ColumnFamily, Error> {
+        self.inner.cf_handle(cf_str).ok_or(Error::UnknownColumnFamily(cf_str))
+    }
+}
+
+#[async_trait]
+impl StorageBackend for Storage {
+    type ConfigBuilder = RocksDbConfigBuilder;
+    type Config = RocksDbConfig;
+    type Error = Error;
+
+    async fn start(config: Self::Config) -> Result<Self, Self::Error> {
+        let storage = Storage {
+            config: config.storage.clone(),
+            inner: Self::new(config)?,
+        };
+
+        match Fetch::<u8, System>::fetch(&storage, &SYSTEM_VERSION_KEY).await? {
+            Some(System::Version(version)) => {
+                if version != STORAGE_VERSION {
+                    return Err(Error::VersionMismatch(version, STORAGE_VERSION));
+                }
+            }
+            None => {
+                Insert::<u8, System>::insert(&storage, &SYSTEM_VERSION_KEY, &System::Version(STORAGE_VERSION)).await?
+            }
+            _ => panic!("Another system value was inserted on the version key."),
+        }
+
+        if let Some(health) = storage.get_health().await? {
+            if health != StorageHealth::Healthy {
+                return Err(Self::Error::UnhealthyStorage(health));
+            }
+        }
+
+        storage.set_health(StorageHealth::Idle).await?;
+
+        Ok(storage)
+    }
+
+    async fn shutdown(self) -> Result<(), Self::Error> {
+        self.set_health(StorageHealth::Healthy).await?;
+
+        Ok(self.inner.flush()?)
+    }
+
+    async fn size(&self) -> Result<Option<usize>, Self::Error> {
+        let mut size = 0;
+        let files = self.inner.live_files()?;
+
+        for file in files {
```

Comment:
maybe `files.iter().map(|file| file.size).sum()` is more concise?

Reply:
`Ok(Some(self.inner.live_files()?.iter().fold(0|acc, file|acc+file.size)))` :trollface: